packages/engine/lib/error/src/macros.rs

```diff
@@ -9,14 +9,31 @@ pub mod __private {
 
         use crate::{Context, Message, Report};
 
+        pub trait ReportTag {
```

Comment:
I think we should add comments on these explaining what they're doing and why they exist. It seems like dark magic.

Reply:
I see you've added the short module doc and a TODO, cheers

Reply:
The link should describe the context well enough, will however explain it a little bit more in-depth in the module to avoid only linking a blog post, which may be gone at one point.